packages/data-stores/src/site/reducer.ts

```diff
@@ -38,25 +43,40 @@ const newSiteError: Reducer< NewSiteErrorResponse | undefined, Action > = ( stat
 	return state;
 };
 
-const isFetchingSite: Reducer< boolean | undefined, Action > = ( state = false, action ) => {
+export const isFetchingSite: Reducer< boolean | undefined, Action > = ( state = false, action ) => {
```

Comment:
I normally try to avoid writing `isFetching...` reducers, as it means I have to manually track a request (and remember to listen to various actions) for something that I should be able to infer differently.

Are you familiar with `@wordpress/core-data`'s [`isResolving` selector](https://github.com/WordPress/gutenberg/blob/1e8d78189a9b05628b171016e102e2c50e267c39/packages/data/src/namespace-store/metadata/selectors.js#L55-L67)? I'm using it e.g. [here](https://github.com/Automattic/wp-calypso/blob/054a6dda3bdbe83f1c482df782f4c885dffc2b63/packages/data-stores/src/domain-suggestions/selectors.ts#L29-L39).

Reply:
(Since it's pre-existing, we can definitely tackle this in a follow-up PR; just wanted to bring it up.)

Reply:
Thanks for bringing this one up — the `isResolving` selector's great for selectors with a matching resolver, but I couldn't find an existing equivalent for actions / controls (I might have missed something obvious!). In this case the `isFetchingSite` reducer is updated by [the `fetchNewSite` action](https://github.com/automattic/wp-calypso/blob/543ce32424996ab799ce391bfbf8642ab1f09a2c/packages/data-stores/src/site/actions.ts#L30), which is fired off at the beginning of the `createSite` generator to flag that creating a site is in a loading state. Perhaps it needs a different name, too? I couldn't quite work out a better approach when I originally put it together.

Reply:
Ah yeah, that's a very good point that I missed :sweat_smile: Gutenberg really could use some helper for that case :thinking: